### ISSUE TITLE
Question/Feature: Ability to Define values on sunset/sunrise

### ISSUE THREAD

**Issue by kirbo** [basnijholt/adaptive-lighting#151]
Hi,
I just noticed last night that my lights are at 100% when the sun is above horizon. I used to use Circadian Lighting previously but it didn't work after some HA update, therefore I decided to switch into Adaptive Lighting.
I've just noticed the difference that with Circadian Lighting, the lights/color temperatures were something like 50% once the sun was just above the horizon and not 100% as it is now.

Is there a way for me to change the configuration in a way that the light wouldn't turn on at 100% at 04am in the morning (when the sun is just rising above horizon), as it is too bright.
I know there are offsets for both sunrise and sunset, but I think this wouldn't be the best practice to handle this.

Therefore I'm proposing that use could define the range at when the lights should go with minimum and maximum values, something like:
```yaml
maximum_elevation: 80
minimum_elevation: -80
```
..meaning that when the sun elevation is 80 or above, it would turn the lights to maximum values (e.g. 100% and 5000k) and when the sun elevation is -80 or below, it would turn the lights to minimum values (e.g. 10% and 2000k).

Perhaps the elevation wouldn't be the way to go either, since it varies by seasons, but I think you get the point I'm trying to make..

At least as a workaround I could just use the offsets, but I still think that wouldn't perhaps be the best way to handle this (or at least it wouldn't be as flexible from users perspective, because here in Finland at summer in the north the sun doesn't set at all (meaning that the lights are full 100% at all times) for a few days and at winter it never rises above the horizon)

**Comment by RubenKelevra** [comment]
Hey @kirbo,

I think the thing you're experiencing is the same I did [here](https://github.com/basnijholt/adaptive-lighting/issues/133). The curve for the brightness isn't really what I would expect and what's natural.

I think It would be nice to be able to choose some predetermined curves, which are more natural.

Would this work for you?

**Comment by kirbo** [comment]
This is what it looks like for me. From 04:20 (am) until 22:39 (10:39 pm) the light is at 100%
![image](https://user-images.githubusercontent.com/1077974/127750262-cb21098a-7a67-4913-b7a6-24ff531ab8e6.png)
![image](https://user-images.githubusercontent.com/1077974/127750265-10070a69-b80d-4534-837f-f4c0d6e9609a.png)
![image](https://user-images.githubusercontent.com/1077974/127750269-23b48489-c0d4-4896-aa5f-20a2dafaec09.png)

**Comment by kirbo** [comment]
> I think It would be nice to be able to choose some predetermined curves, which are more natural.
>
> Would this work for you?

**Comment by kirbo** [comment]
...OOOOOR, could this integration check when the sunrises and sets (calculates how long the day lasts/how long the sun is above horizon) and calculates for example 80% of the time when the lights would be at maximum level.

For example, if the sunrise at 09 and sunset at 21 (sun above horizon for 12h), then it would calculate 12h * 80% = 9.6h, which would mean that the lights would be at 100% from 10:36 until 19:24, or something like that. My calculations might not be right, but that was just to give you an example what I'm trying to think 😅

**Comment by RubenKelevra** [comment]
> What if user could define it's own sunset/sunrise values, that would be the earliest/latest? I mean, something like:
> 
> * sunrise
>   
>   * min: 06:00
>   * max: 10:00
> * sunset
>   
>   * min: 20:00
>   * max: 23:00
> 
> Meaning that the day would vary from 06-23 into 10-20, so the variation between day lengths would be 10h into 17hours.
> 
> I mean, that this integration would ignore the sunrise, if the sunrises before 06 and after 10 and ignores the sunset before 20 and after 23.
> 
> But I mean that these "sunrise min/max, sunset min/max" should be user definable.
> 
> ..just some random thoughts how this could perhaps be solved..

That has already be requested and is tracked here: https://github.com/basnijholt/adaptive-lighting/issues/97

**Comment by RubenKelevra** [comment]
> ...OOOOOR, could this integration check when the sunrises and sets (calculates how long the day lasts/how long the sun is above horizon) and calculates for example 80% of the time when the lights would be at maximum level.
> 
> For example, if the sunrise at 09 and sunset at 21 (sun above horizon for 12h), then it would calculate 12h * 80% = 9.6h, which would mean that the lights would be at 100% from 10:36 until 19:24, or something like that. My calculations might not be right, but that was just to give you an example what I'm trying to think 😅
> 
> Or easier: sunrise 10 and sunset 20, the sun would be above horizon for 10h so the lights would be at maximum level for 8 hours, which would mean from 11 until 19.
> And same if the sunrise at 12 and sunset at 17, above horizon for 5h, lights would be at maximum level for 4hours, from 12:30 until 16:30.

Yeah that's basically what I meant with a more realistic curve :)

**Comment by RubenKelevra** [comment]
@kirbo so this can be closed as duplicate? :)

**Comment by kirbo** [comment]
> @kirbo so this can be closed as duplicate? :)

Fine by me
